### DISCUSSION TITLE
rook mgr module disabled by default?

### DISCUSSION THREAD

**Discussion by sabaini** [rook/rook#11587] [Q&A]
It appears that by default the rook mgr module is disabled by default.

```
bash-4.4$ ceph mgr module ls
MODULE                           
...
rook               -             
```

This in turn leads causes the `ceph orch backend` to be unset which makes the dashboard unhappy in some places.

Deploying with the below enables the module and causes rook to set the backend

```
mgr:
    modules:
      - name: rook
        enabled: true
```

Would it make sense to have the rook module enabled-by-default (in turn setting the orch backend by default)?

cheers,
peter.

Rook v1.10.6
ceph version 17.2.5 (98318ae89f1a893a6ded3a640405cdbb33e08757) quincy (stable)

**Comment by subhamkrai** [comment]
@sabaini IIRC, some work are required on the rook side to enable the mgr module by default. Hopefully, soon it will be enabled by default.

**Reply by travisn** [reply]
The rook mgr module was disabled in #5834 due to some issues it was causing with errors in the logs, or reliability in the CI. Agreed we need to resolve those issues and get it enabled again by default for the dashboard to work fully. @jmolmo Could you look at enabling it again by default?

**Reply by travisn** [reply]
We may also need to only enable it by default for the latest version of ceph with some recent rook mgr module fixes.

**Reply by jmolmo** [reply]
In order to enable the rook manager module we need first to have integration tests running. I am working on this part to have a jenkins job in the Ceph project to test the rook orchestrator functionality. I hope to have a first version running very soon.
Once we have IT, we can safely enable the rook orchestrator.
